### ISSUE TITLE
DB: Some channels have no source_id or xmltv_id

### ISSUE THREAD

**Issue by billmeek** [MythTV-Clients/MythTV-Android-Frontend#201]
The following shows the channel table is is storing a
source id of 0 and or xmltv_id of '' for some channels.

Thinking that if the XMLTVID is the same for channels
that exist on two or more sources, then the channel
should be displayed in the guide once. Not sure how
that works in the rest of the world.

``` sql
select chan_id,chan_num,callsign,source_id,xmltv_id
    from channel where source_id < 1
        or xmltv_id = '' order by chan_id;
1021        2_1         CBS2-HD     0                     
1051        5_1         NBC5        0                     
1071        7_1         WLS-HD      0                     
1111        11_1        WTTW-HD     0                     
1112        11_2        WTTW-DT     0                     
1201        20_1        WYCC HD     0                     
1321        32_1        WFLD        0                     
2117        117         COMEDY      0                     
2142        142         AMC         0                     
2143        143         TCM         0                     
2225        225         HGTV        0                     
3623        623         TNTHD       0                     
3630        630         SYFYHD      0                     
3632        632         CCHD        0
```

There are no such channels in the MythTV DB

``` sql
select chanid,callsign,sourceid,xmltvid from channel
    where visible >  0 and (sourceid < 1 or xmltvid = '')
        order by chanid;

Empty set (0.00 sec)
```

And, looking at just one channel's Channel/GetChannelInfoList?SourceID=1,
the XMLTVID and sourceid are properly set. No channels were returned for
SourceID=0.

``` xml
<ChannelInfo>
<ChanId>1021</ChanId>
<ChanNum>2_1</ChanNum>
<CallSign>CBS2-HD</CallSign>
<IconURL>wbbm_cbs2_chicago.jpg</IconURL>
<ChannelName>CBS2-HD</ChannelName>
<MplexId>1</MplexId>
<TransportId>999</TransportId>
<ServiceId>1</ServiceId>
<NetworkId>0</NetworkId>
<ATSCMajorChan>2</ATSCMajorChan>
<ATSCMinorChan>1</ATSCMinorChan>
<Format>ATSC</Format>
<Modulation>8vsb</Modulation>
<Frequency>207000000</Frequency>
<FrequencyId>12</FrequencyId>
<FrequencyTable>default</FrequencyTable>
<FineTune>0</FineTune>
<SIStandard>atsc</SIStandard>
<ChanFilters/>
<SourceId>1</SourceId>
<InputId>0</InputId>
<CommFree>0</CommFree>
<UseEIT>false</UseEIT>
<Visible>true</Visible>
<XMLTVID>20454</XMLTVID>
<DefaultAuth/>
<Programs/>
</ChannelInfo>
```

**Comment by dmfrey** [comment]
The channels get updated when the recorded episodes are update so as to add any missing channels (i.e. you have a Miro channel setup for Miro Bridge, but not tied to a VideoSource.  I am refining that logic to only update if it doesn't exist.

For some reason, the recorded list comes back with channels, but not all the data is populated

**Comment by billmeek** [comment]
Much better. There is only one error and that's 2143 which shouldn't
be there:

``` sql
sqlite> .mode column
sqlite> .width 10 10 10 10 10
sqlite> select chan_id,chan_num,callsign,source_id,xmltv_id
    from channel where source_id < 1
        or xmltv_id = '' order by chan_id;

2143        143         TCM         0 
```

I'll leave this closed and track it in #190.
